From 099ee7b958182e4ae59d42c4492ea171b2d5cc20 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Tue, 12 Dec 2017 10:08:03 +0900
Subject: [PATCH 385/909] media: i2c: adv7180: Add std update when getting
 format

commit adbfc9e75545e1577e0fc20f59afd120c9e4b481 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Signal information could not be acquired correctly when input
signal was switched. This problem is solved by updating
std (Standard Definition) at the get_fmt function.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/i2c/adv7180.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/i2c/adv7180.c b/drivers/media/i2c/adv7180.c
index 25d24a3f10a7..3633f8bbc385 100644
--- a/drivers/media/i2c/adv7180.c
+++ b/drivers/media/i2c/adv7180.c
@@ -3,6 +3,7 @@
  * Copyright (c) 2009 Intel Corporation
  * Copyright (C) 2013 Cogent Embedded, Inc.
  * Copyright (C) 2013 Renesas Solutions Corp.
+ * Copyright (C) 2017 Renesas Electronics Corp.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
@@ -690,10 +691,13 @@ static int adv7180_get_pad_format(struct v4l2_subdev *sd,
 				  struct v4l2_subdev_format *format)
 {
 	struct adv7180_state *state = to_state(sd);
+	v4l2_std_id std = 0;
 
 	if (format->which == V4L2_SUBDEV_FORMAT_TRY) {
 		format->format = *v4l2_subdev_get_try_format(sd, cfg, 0);
 	} else {
+		adv7180_querystd(sd, &std);
+		state->curr_norm = std;
 		adv7180_mbus_fmt(sd, &format->format);
 		format->format.field = state->field;
 	}
-- 
2.17.1

